### PR TITLE
chore(deps): remove deprecated @types/strip-ansi

### PR DESCRIPTION
**Pull request by MikeMcC399**
- Contributes to clearing Renovate's deprecation list on https://github.com/cypress-io/cypress/issues/3777

### Additional details

[packages/errors](https://github.com/cypress-io/cypress/tree/develop/packages/errors) specifies [strip-ansi@6.0.1](https://github.com/chalk/strip-ansi/releases/tag/v6.0.1).

https://github.com/cypress-io/cypress/blob/58abe803f68e95c5d35fc385c079d28e4b8994bc/packages/errors/package.json#L22

Specifying `@types/strip-ansi@5.2.1` in addition to [strip-ansi@6.0.1](https://github.com/chalk/strip-ansi/releases/tag/v6.0.1)is incorrect since [strip-ansi@5.2.0](https://github.com/chalk/strip-ansi/releases/tag/v5.2.0) already introduced a TypeScript definition. Leaving `@types/strip-ansi@5.2.1` in `devDependencies` could potentially result in conflicts. Also the npm module [@types/strip-ansi](https://www.npmjs.com/package/@types/strip-ansi) is deprecated.

https://github.com/cypress-io/cypress/blob/58abe803f68e95c5d35fc385c079d28e4b8994bc/packages/errors/package.json#L31

This PR removes the deprecated and redundant `@types/strip-ansi@^5.2.1` specification from [packages/errors/package.json](https://github.com/cypress-io/cypress/blob/develop/packages/errors/package.json).

### Steps to test

```shell
yarn
```

### How has the user experience changed?

No change to user experience.

### PR Tasks

- [na] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by MikeMcC399** [comment]
TypeScript tests passed. One test `server-integration-tests` failed.

**Comment by MikeMcC399** [comment]
On re-run all tests from first batch pass. Now waiting for CircleCI approval to run second batch of CI tests.

**Review by jennifer-shehane — APPROVED** [review]

**Review by mschile — APPROVED** [review]

**Pull request merged**
